qiskit/circuit/library/generalized_gates/diagonal.py

```diff
@@ -121,3 +122,20 @@ def _extract_rz(phi1, phi2):
     phase = (phi1 + phi2) / 2.0
     z_angle = phi2 - phi1
     return phase, z_angle
+
+
+# Returns size of Diagonal or 2^k
+def _exp_two(n: int) -> int:
+    i = 1
+    while i <= n:
+        i *= 2
+    return i
```

Comment:
Isn't this just `2 ** n`?

Reply:
No, `_exp_two` returns nearest power of 2 to the `n` (right hand limit). for instance `_exp_two(5)` will return 8.